libafl_qemu/src/qemu/systemmode/intel_pt.rs

```diff
@@ -0,0 +1,169 @@
+use std::{
+    fs,
+    os::{
+        fd::{AsRawFd, FromRawFd, OwnedFd},
+        raw::c_void,
+    },
+};
+
+use bitflags::bitflags;
+use libc::ioctl;
+use perf_event_open_sys::{
+    bindings::{perf_event_attr, perf_event_mmap_page, PERF_FLAG_FD_CLOEXEC},
+    perf_event_open,
+};
+
+const PAGE_SIZE: usize = 4096;
+const PERF_BUFFER_SIZE: usize = ((1 + (1 << 7)) * PAGE_SIZE) as usize;
+const PERF_AUX_BUFFER_SIZE: usize = 64 * 1024 * 1024;
+const PT_EVENT_TYPE_PATH: &str = "/sys/bus/event_source/devices/intel_pt/type";
+
+bitflags! {
+    /// IA32_RTIT_CTL MSR flags
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    struct PtConfig: u64 {
+        /// Disable call return address compression.
+        /// AKA DisRETC in Intel SDM
+        const NORETCOMP = 0b1 << 11;
+    }
+}
+
+// TODO consider moving to static assert crate
+const _: () = assert!(
+    ((PERF_BUFFER_SIZE - PAGE_SIZE) & (PERF_BUFFER_SIZE - PAGE_SIZE - 1)) == 0,
+    "PERF_BUFFER_SIZE should be 1+2^n pages"
+);
+const _: () = assert!(
+    (PERF_AUX_BUFFER_SIZE % PAGE_SIZE) == 0,
+    "PERF_AUX_BUFFER_SIZE must be page aligned"
+);
+const _: () = assert!(
+    (PERF_AUX_BUFFER_SIZE & (PERF_AUX_BUFFER_SIZE - 1)) == 0,
+    "PERF_AUX_BUFFER_SIZE must be a power of two"
+);
+// Ensure we can use usize and u64 interchangeably
+const _: () = assert!(
```

Comment:
i'd suggest to use `static_assertions`, it's already a depency of `libafl_bolts` and it's more readable.

Reply:
Ok done, thanks!
The only downside is that their macros do not support custom error messages like `assert!(false,"Custom error message");`

Reply:
I'd say having a comment above the assert is good enough, ppl can look at the code if that triggers